waku/v2/protocol/store/waku_store_protocol_test.go

```diff
@@ -60,7 +60,9 @@ func TestWakuStoreProtocolQuery(t *testing.T) {
 		ContentTopics: []string{topic1},
 	}
 
-	response, err := s2.Query(ctx, q, DefaultOptions()...)
+	var hrOptions []HistoryRequestOption
+	hrOptions = append(hrOptions, WithPeer(host1.ID()))
+	response, err := s2.Query(ctx, q, hrOptions...)
```

Comment:
Yes, because now peermanager is required for peerSelection it is difficult to write unit tests at each protocol level for these. Unit tests at peerManager level validate automatic PeerSelection, just that at each protocol level we are not able to have tests.
Maybe this can be addressed by having such tests at WakuNode level.
Will create a separate issue to add tests for this to analyze and add.